src/Cli/dotnet/Commands/Solution/Add/SolutionAddCommand.cs

```diff
@@ -173,7 +175,7 @@ private void AddProject(SolutionModel solution, string fullProjectPath, ISolutio
             Reporter.Error.WriteLine(CliStrings.UnsupportedProjectType, fullProjectPath);
             return;
         }
-        catch (SolutionArgumentException ex) when (ex.Type == SolutionErrorType.DuplicateProjectName || solution.FindProject(solutionRelativeProjectPath) is not null)
+        catch (SolutionArgumentException ex) when (showMessageOnDuplicate && (ex.Type == SolutionErrorType.DuplicateProjectName || solution.FindProject(solutionRelativeProjectPath) is not null))
```

Comment:
Doesn't this just mean that now it throws when it finds a duplicate? I would've expected it to go in an if under this rather than in the when clause.

Reply:
Ohh yes, you're right.